SAMLPortal/Controllers/SetupController.cs

```diff
@@ -0,0 +1,268 @@
+			model.BindDN = ldapBindDn != null ? ldapBindDn : "";
+			model.BindPassword = ldapBindPassword != null ? ldapBindPassword : "";
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("3")]
+		public IActionResult StepThree(StepThreeModel model)
+		{
+			if (ModelState.IsValid)
+			{
+				LdapConnection connection = new LdapConnection();
+				connection.SecureSocketLayer = model.SSL;
+
+				try
+				{
+					connection.Connect(model.Host, model.Port);
+					connection.Bind(model.BindDN, model.BindPassword);
+
+					//If everything goes well
+					GlobalSettings.Store("LDAP_Host", model.Host);
+					GlobalSettings.Store("LDAP_Port", model.Port.ToString());
+					GlobalSettings.Store("LDAP_SSL", model.SSL.ToString());
+					GlobalSettings.Store("LDAP_BindDN", model.BindDN);
+					GlobalSettings.Store("LDAP_BindPass", model.BindPassword);
+
+					Helpers.ReplaceEnvVariableInFile(GlobalSettings.Get("CONFIG_FILE"), "SP_CONFIG_SETUPASSISTANT_STEP", "4");
+					GlobalSettings.Store("CONFIG_SETUPASSISTANT_STEP", "4");
+					return Redirect("4");
+
+				}
+				catch (LdapException ex)
+				{
+					ModelState.AddModelError(string.Empty, ex.Message);
+				}
+				catch (AggregateException ex)
+				{
+					ModelState.AddModelError(string.Empty, ex.Message);
+				}
+				catch (Exception ex)
+				{
+					ModelState.AddModelError(string.Empty, "An unknown error occurred. Please try again.");
+					Console.WriteLine(ex.Message);
+				}
+			}
+
+			return View(model);
+		}
+
+		[HttpGet]
+		[Route("4")]
+		public IActionResult StepFour()
+		{
+			var ldapSearchBase = GlobalSettings.Get("LDAP_SearchBase");
+			var ldapAdminFilter = GlobalSettings.Get("LDAP_AdminFilter");
+			var ldapUserFilter = GlobalSettings.Get("LDAP_UsersFilter");
+
+			StepFourModel model = new StepFourModel();
+			model.SearchBase = ldapSearchBase != null ? ldapSearchBase : "";
+			model.AdminFilter = ldapAdminFilter != null ? ldapAdminFilter : "uid={0}";
+			model.UserFilter = ldapUserFilter != null ? ldapUserFilter : "uid={0}";
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("4")]
+		public IActionResult StepFour(StepFourModel model)
+		{
+			if (ModelState.IsValid)
+			{
+				GlobalSettings.Store("LDAP_SearchBase", model.SearchBase);
+				GlobalSettings.Store("LDAP_AdminFilter", model.AdminFilter);
+				GlobalSettings.Store("LDAP_UsersFilter", model.UserFilter);
+
+				Helpers.ReplaceEnvVariableInFile(GlobalSettings.Get("CONFIG_FILE"), "SP_CONFIG_SETUPASSISTANT_STEP", "5");
+				GlobalSettings.Store("CONFIG_SETUPASSISTANT_STEP", "5");
+			}
+
+			return View(model);
+		}
+	}
+
+	public class SetupAsyncActionFilter : IAsyncActionFilter
+	{
+		public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+		{
+			if (GlobalSettings.GetInt("CONFIG_SETUPASSISTANT_STEP") > 5)
+			{
+				// Setup has been completed and cannot be accessed anymore
+				context.HttpContext.Response.Redirect("/");
+			}
+
+			var resultContext = await next();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'resultContext'.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)